pkg/providers/docker/controlplane.go

```diff
@@ -0,0 +1,96 @@
+package docker
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	dockerv1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	yamlcapi "github.com/aws/eks-anywhere/pkg/clusterapi/yaml"
+	"github.com/aws/eks-anywhere/pkg/yamlutil"
+)
+
+// ControlPlane represents a CAPI Docker control plane.
+type ControlPlane = clusterapi.ControlPlane[*dockerv1.DockerCluster, *dockerv1.DockerMachineTemplate]
+
+// ControlPlaneBuilder defines the builder for all objects in the CAPI Docker control plane.
+type ControlPlaneBuilder struct {
```

Comment:
Do we need this struct? I think we can also remove this since the docker provider has no extra cp objects. The builder that it creates should suffice

Reply:
Agreed. Tis cleaner